src/nspawn/nspawn-network.c

```diff
@@ -743,3 +750,53 @@ int remove_veth_links(const char *primary, char **pairs) {
 
         return 0;
 }
+
+static int network_iface_pair_parse_and_test(const char* iftype, char ***l, const char *p, const char* ifprefix) {
+        _cleanup_free_ char *a = NULL, *b = NULL;
+        int r;
+
+        r = extract_first_word(&p, &a, ":", EXTRACT_DONT_COALESCE_SEPARATORS);
+        if (r < 0)
+                return log_error_errno(r, "Failed to extract first word in %s parameter: %m", iftype);
+        if (r == 0)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "Short read while reading %s parameter: %m", iftype);
+        if (!ifname_valid(a))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "%s, interface name not valid: %s", iftype, a);
+
+        r = test_network_interface_initialized(a);
+        if (r < 0)
+                return r;
+
+        if (!p || isempty(p)) {
+                b = strjoin(ifprefix, a);
+        } else {
+                b = strdup(p);
+        }
+        if (!b)
+                return log_oom();
+
+        if (!ifname_valid(b))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                        "%s, interface name not valid: %s", iftype, b);
+
+        r = strv_push_pair(l, a, b);
+        if (r < 0)
+                return log_oom();
+
+        a = b = NULL;
+        return 0;
+}
+
+int interface_pair_parse_and_test(char ***l, const char *p) {
+        return network_iface_pair_parse_and_test("Network interface", l, p, "");
```

Comment:
you handle a NULL ifprefix just fine above, hence pass NULL as third arg here

Reply:
:+1: 
`network_iface_pair_parse_and_test` changed to support third arg=NULL.